File: go/rna-transcription/rna_transcription.go
package strand

const testVersion = 3

var translations = map[rune]rune{
	'G': 'C',
	'C': 'G',
	'T': 'A',
	'A': 'U',
}

// ToRNA converts a sequence of dna to its rna complement.
func ToRNA(dna string) string {
	sequence := []rune(dna)

	for i, d := range sequence {
		sequence[i] = translations[d]
	}

	return string(sequence)
}

File: ruby/rna-transcription/rna_transcription.rb
# Transcribe a DNA strand into it's complement.
module Complement
  DNA = 'GCTA'.freeze
  RNA = 'CGAU'.freeze

  def self.of_dna(strand)
    return '' unless strand.delete(DNA).empty?
    strand.tr(DNA, RNA)
  end
end

module BookKeeping
  VERSION = 4
end
